### ISSUE TITLE
Tests failing for package_coverage but not R CMD CHECK on TravisCI

### ISSUE THREAD

**Issue by epurdom** [r-lib/covr#320]
Hello,
I have a package `clusterExperiment` that uses TravisCI to check the package and run `coveralls`. I have the problem that the package runs cleanly on TravisCI, but that the `package_coverage` hits an error in my tests. 

This is also the case when I run it locally, and when I run it locally I can look at the error that `package_coverage`  is hitting:
```
── 1. Failure: check clusterLegend manipulations work as promised (@test_structu
`cc` not equal to `newcc`.
Attributes: < Component “transformation”: target, current do not match when deparsed >

══ testthat results  ═══════════════════════════════════════════════════════════
OK: 399 SKIPPED: 1 FAILED: 1
1. Failure: check clusterLegend manipulations work as promised (@test_structural_constructor.R#291) 

Error: testthat unit tests failed
Execution halted
```
(I don't actually know that it is the same error as TravisCI, since the output of package_coverage doesn't print out the end of the errored file, but only the beginning where it's loading the packages).

This test passes TravisCI, and when I test it locally on my machine. I'm not sure why its getting an error in `package_coveralls`. 

The github for the package is `epurdom/clusterExperiment`. And while I only recently realized the problem, the package has been failing coveralls on TravisCI for a long-time now (since February I think) despite the fact that it has been passing in both TravisCI and on Bioc.

Thanks for any assistance in figuring this out. 
Elizabeth Purdom
@drisso

**Comment by rubak** [comment]
In `spatstat` <https://github.com/spatstat/spatstat> we have just encountered a similar issue. It seems to be the combination of R 3.4.4 which runs on TravisCI and the new 3.1.0 version of `covr` which came out in May. When we install the old 3.0.1 version of `covr` everything runs smoothly. I *think* @baddstats ran the same tests on R 3.5 with `covr` 3.1.0 without any issues. Are you able to reproduce similar results in your case with different versions of R/`covr`? In that case it might be worth looking into the issue of version conflicts and make a proper bug report for the `covr` maintainers.

Note: you can easily install the old version of `covr` with `devtools::install_version("covr", "3.0.1")`

**Comment by maxheld83** [comment]
I had what appears to be a similar issue with our package [pensieve](http://github.com/maxheld83/pensieve), where some test *passed* `devtools::test()` (also on Travis), but *failed* `covr::codecov()` on Travis.

The source of the issue might have well been a legitimate bug on our side, so it's not clear that this is related to the above issues, but I thought I'd add this in case it is helpful to track down the inconsistency between `covr::codecov()` and `devtools::test()`, which we also observed in our case.

Our error message from [this travis build, deploy stage](https://travis-ci.com/maxheld83/pensieve/jobs/139263364):

```
[0K$ Rscript -e 'covr::codecov()'
Error: Failure in `/tmp/Rtmp1f8UKn/R_LIBSb038c998d7b/pensieve/pensieve-tests/testthat.Rout.fail`
> library(testthat)
> library(pensieve)

Attaching package: â€˜pensieveâ€™

The following object is masked from â€˜package:testthatâ€™:

    expect

> 
> test_check("pensieve")
â”€â”€ 1. Error: from/to R objects works (@test_markup2vector.R#110)  â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€â”€
`x` must be a character
1: svg2grob_mem(x = virt_svg) at testthat/test_markup2vector.R:110
2: withVisible(`_f`(x = x, path_in = path_in, ... = ...))
3: `_f`(x = x, path_in = path_in, ... = ...)
4: is_binary(path_out)
5: fs::path_ext(path) %in% c("md", "tex", "txt")
6: fs::path_ext(path)
7: captures(path, regexpr("(?<!^|[.]|/)[.]([^.]+)$", path, perl = TRUE))
8: assert("`x` must be a character", is.character(x))
```

We fixed the issue with [this diff](https://github.com/maxheld83/pensieve/compare/f5f88a75ec72...a2dc87fc3d63).

My very preliminary hunch is that somehow, the functional operator in the above diff was compiled differently by whatever build command `covr::codecov()` used compared to what `devtools::test()` does before running the tests.
In the latter case, the (iffy) code worked, in the former not.

Again, we solved the problem, but it was weird that this would trip up *only*  `covr::codecov()` and none of the other test tools or environments.

**Comment by maxheld83** [comment]
thanks @jimhester and sorry for the false track then.

> Also in general comparing in your package with other functions for equality will likely fail with covr, which works by modifying the function definition (to add tracing statements) in your package functions.

This explains our observation, makes perfect sense.

And yes, that was bad code on our part anyway.

**Comment by nlneas1** [comment]
I _think_ that this may have something to do with `covr` not running the `setup-` testthat scripts. I used `package_coverage()` and I was getting some odd failures. I tried running it with `quiet  = FALSE` and `clean = FALSE` but this didn't really help much. I then started using `type = "none"` and running my `setup.R` file code within the `code` parameter followed by my tests and they all of a sudden started working.
